onadata/libs/utils/middleware.py

```diff
@@ -64,3 +75,20 @@ def __call__(self, request):
                     query['time'], " ".join(query['sql'].split())))
 
         return response
+
+
+class OperationalErrorMiddleware(BaseMiddleware):
+    """
+    Captures requests returning 500 status code.
+    Then retry it against master database.
+    """
+    def process_exception(self, request, exception):
+        # Filter out OperationalError Exceptions
+        if isinstance(exception, OperationalError):
+            already_raised = getattr(settings, "ALREADY_RAISED", False)
```

Comment:
I believe keeping this value in the settings file means all subsequent errors after the first will be ignored as this is not tracking independent requests.

Reply:
I'm not sure if the setting traverses across multiple threads, an attempt to use `threading.local()` did not work. Using settings is what seemed to work.

Reply:
I am not sure we should add the setting `ALREADY_RAISED` to the setting files, it should just have life in this middleware in the event it is used. Using `getattr()` with a default value is a valid use case with no required configuration.

Reply:
I agree. `ALREADY_RAISED` variable should be defined and only used in this middleware.